### PR TITLE
Update

### PR DESCRIPTION
**Pull request by ssemugenyi**
# Davsafaris changes

### Manage countries(new)
Added creating, editing, deleting a new country, and fetching all countries
Refetching countries after creating a new country
### Manage categories(new)
Added creating, editing, deleting a new country, and fetching all categories
Refetching category after creating a new category
### Manage tours
Fetching countries and categories dynamically for each tour
Refetching tours after creating a new tour

## Navigation Menu
Added other destinations  dropdown  with a slider 
Responsiveness of the navigation on all devices

## Homepage
Fetching all countries and populating them in the search component
Added a carousel component for all countries created

## Country Single and Category Single Page
Fetched country and category by slug
Added a skeleton to the page before loading content

### CODE REVIEW

**Comment by KimeraMoses** [comment]
image

This section still has dummy countries, can you please make them dynamic with countries with most number of tours

**Comment by KimeraMoses** [comment]
image

Can we eliminate any country with zero countries from the list in the agent dashboard, it doesn't make sense for us to show them a country with no tours

**Comment by KimeraMoses** [comment]
image

Make uganda to be the default of this select by county dropdown since all the countries displayed initially are for Uganda

File: src/components/DashBoard/ManageCountries/EditCountry.js [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+                      label="Country Specialist"
+                      variant="filled"
+                      name="specialist"
+                      value={values.specialist}
+                      onChange={onChangeHandler}
+                      className={styles.gpa__register_form_right_wrapper}
+                    />
+                  </div>
+                  <div className="col-xs-12 col-sm-6">
+                    <TextField
```
Comment (KimeraMoses) [inline src/components/DashBoard/ManageCountries/EditCountry.js:313]: Completely remove this section please

File: src/components/DashBoard/ManageCountries/delete/DeleteModal.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  const style = {
+    bgcolor: "background.paper",
+  };
+
+  const deleteHandler = async () => {
+    setLoading(true);
+    if (source === "country") {
+      await DAV_APIS.deleteCountryById(Id);
+      toast.success("Country deleted Successfully");
+      // dispatch(fetchAllCountrys());
```
Comment (KimeraMoses) [inline src/components/DashBoard/ManageCountries/delete/DeleteModal.js:33]: Remove any redux related code that we have commented out and not using. This includes the slices, actions and any other file or line of code that you are sure that we are not using any more

**Review by KimeraMoses — CHANGES_REQUESTED** [review]
Please fix these issues and those raised in the issue section and let me know when you have pushed the code

**Comment by KimeraMoses** [comment]
image
Remove this field from this form and from anyother form that you worked on, we don't need it visible at all

**Comment by KimeraMoses** [comment]
image
These two fields are quite confusing, can you please change the labels to something less confusing. I'm sure admins will be asking what we mean by title and name

**Comment by KimeraMoses** [comment]
image
This is how the live Kenya version looks like and below is how the new version looks like, question:

image

Where are these categories, once deployed, we shall not be viewing these categories if not created, we need to create them for any country that had them ensuring that nothing gets broken in the middle of the process.

File: src/components/CountryTours/CountryHeader.js [hunk 3]
```diff
@@ -1,9 +1,18 @@
...
 import classes from "./CountryHeader.module.css";
+// import { useSelector } from "react-redux";
 
 import { Link } from "react-router-dom";
+// import { selectAllCategories } from "../../store/Slices/fetchCategoriesSlice";
+import { useAllCategories } from "../../hooks";
 
 const CountryHeader = (props) => {
+  // const fetchedCategories = useSelector(selectAllCategories);
+  // let categories = fetchedCategories.categories;
```
[the remark this answers is not in the document]
Comment (ssemugenyi) [reply src/components/CountryTours/CountryHeader.js:11]: Okay am going to fix them, let me wait for Isaac to first create the popular destinations endpoint

**Comment by KimeraMoses** [comment]
image
The reload button doesn't work for the safari updates and language posts, could you have an idea why this is like this?

**Comment by KimeraMoses** [comment]
All changes merged to master

**Pull request merged**